sql/api/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -523,6 +523,24 @@ abstract class SparkSession extends Serializable with Closeable {
     sql(sqlText, args.asScala.toMap)
   }
 
+  /**
+   * Executes a SQL query substituting parameters by the given arguments with optional names,
+   * returning the result as a `DataFrame`. This API eagerly runs DDL/DML commands, but not for
+   * SELECT queries. This method allows the inner query to determine whether to use positional
+   * or named parameters based on its parameter markers.
+   *
+   * @param sqlText
+   *   A SQL statement with named or positional parameters to execute.
+   * @param args
+   *   An array of Java/Scala objects that can be converted to SQL literal expressions.
+   * @param paramNames
+   *   An optional array of parameter names corresponding to args. If provided, enables named
+   *   parameter binding where parameter names are available. If None or shorter than args,
+   *   remaining parameters are treated as positional.
```

Comment:
I'm a bit confused here. As of today, we require the parameter markers and the supplied parameter values to match
1. if the query uses named parameter markers,  e.g. `:name`, then the supplied parameter values must be a Map.
2. if the quert uses positional parameter markers, e.g. `?`, then the supplied parameter values must be an Array.

What is the change here? The supplied parameter values are two arrays so that we don't need to care of the parameter markers are named or positional?

Reply:
The  problem is that EXECUTE IMMEDIATE ... USING does NOT have the information:
USING a, b, c can be either unnamed or named as all are variable references (named )and have no explicit alias (unnamed??).
Only when we parse the query in the nested sql() invocation, will we know.
So we need to cater to both options. and that has to be either an array of struct, or two arrays.
I think needs to be solved in analysis where we need to remember state whether we have seen one or the other (or both).

Reply:
In the general case the USING clause could be rand() as p; and p coudl be used multiple times. We need to evaluate the expression.